SEM-lab1/Assets/Scripts/PlayerController.cs

```diff
@@ -1,34 +1,79 @@
 ﻿using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
+using UnityEngine.SceneManagement;
 
 public class PlayerController : MonoBehaviour
 {
     GameController _gameController;
-    public float speed = 5.0f;
+    public float rotateSpeed = 5.0f;
+    public float powerMin = 5f;             
+    public float powerMax = 500f;              
+    public float chargeTime = 5f;     
+    private float _chargeTimeCurr = 0f;
+    private bool _isCharging = false;       
+    private bool _spaceUp = true;
+    private bool _fired = false;
+    private Quaternion _originalRotation;
 
     // Start is called before the first frame update
     void Start()
     {
         _gameController = GameObject.FindGameObjectWithTag("GameController").GetComponent<GameController>();
+        this._originalRotation = this.transform.rotation; //saves initial rotation for refiring so that the ball is not at a strange angle due to rolling
```

Comment:
Don't need to write "this.X", reference to the gameObject is implicit.

Reply:
Confirmed it works without 'this', will change for the next commit.